flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientFactory.java

```diff
@@ -81,14 +92,36 @@ NettyPartitionRequestClient createPartitionRequestClient(ConnectionID connection
 				Object old = clients.putIfAbsent(connectionId, connectingChannel);
 
 				if (old == null) {
-					nettyClient.connect(connectionId.getAddress()).addListener(connectingChannel);
-
-					client = connectingChannel.waitForChannel();
-
-					clients.replace(connectionId, connectingChannel, client);
+					int count = 0;
+					ChannelException exception = null;
+					do {
+						try {
+							exception = null;
+							connectingChannel.error = null;
```

Comment:
Why don't you create a new `ConnectingChannel(...)`?

Reply:
Yes, the code will be cleaner.